specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/apimgateways.json

```diff
@@ -0,0 +1,1015 @@
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The response body contains a collection of Api entities in the gateway.",
+            "schema": {
+              "$ref": "./definitions.json#/definitions/ApiCollection"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./apimanagement.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/gateways/{gatewayId}/apis/{apiId}": {
+      "head": {
+        "tags": [
+          "GatewayApi"
+        ],
+        "operationId": "GatewayApi_GetEntityTag",
+        "description": "Checks that API entity specified by identifier is associated with the Gateway entity.",
+        "x-ms-examples": {
+          "ApiManagementHeadGatewayApi": {
+            "$ref": "./examples/ApiManagementHeadGatewayApi.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "./apimanagement.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ServiceNameParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/GatewayIdParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ApiIdParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Specified Gateway entity exists and current entity state version is present in the ETag header.",
+            "headers": {
+              "ETag": {
+                "description": "Current entity state version. Should be treated as opaque and used to make conditional HTTP requests.",
+                "type": "string"
+              }
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./apimanagement.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "GatewayApi"
+        ],
+        "operationId": "GatewayApi_CreateOrUpdate",
+        "description": "Adds an API to the specified Gateway.",
+        "x-ms-examples": {
+          "ApiManagementCreateGatewayApi": {
+            "$ref": "./examples/ApiManagementCreateGatewayApi.json"
+          }
+        },
+        "parameters": [
```

Comment:
>parameters [](start = 9, length = 10)

how is the user choosing which API is getting added here?  Is it creating a new API?

Reply:
API is already existing resource. User creates Api, Gateway and then assign Api to Gateway

Reply:
This PUT operation does not have a body.  How does the user specify what API they are linking to this gateway?  It should take a body that contains the API's resource ID.

Reply:
API id is in the url: PUT ..../gateways/:gwId/apis/:apiId
We follow this pattern for associations accross all other entities.

Reply:
This pattern isn't aligned with any other api in ARM. All Put requests have a body which contains the definition of what the user is creating. Can we add a body which doesn't have any properties?

Reply:
The definition is still required at least to have provisioning State field

Reply:
@sanjaiganesh, @ryansbenson , @pilor , I added optional body definition with a single provisioningState field. Please have a look.

Reply:
@sanjaiganesh why provisioning state is needed? This is not ARM tracked resource. We already have a lot of such resourced that albeit having a body and schema do not have that property.

Reply:
@sanjaiganesh, @ryansbenson , @pilor
gentle ping